internal/resources/cloud/resource_cloud_stack_service_account_token.go

```diff
@@ -0,0 +1,177 @@
+package cloud
+
+import (
+	"context"
+	"log"
+	"strconv"
+	"time"
+
+	gapi "github.com/grafana/grafana-api-golang-client"
+	"github.com/grafana/terraform-provider-grafana/internal/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func ResourceStackServiceAccountToken() *schema.Resource {
+	return &schema.Resource{
+		Description: `
+**Note:** This resource is available only with Grafana 9.1+.
+
+Manages service account tokens of a Grafana Cloud stack using the Cloud API
+This can be used to bootstrap a management service account token for a new stack
+
+* [Official documentation](https://grafana.com/docs/grafana/latest/administration/service-accounts/)
+* [HTTP API](https://grafana.com/docs/grafana/latest/developers/http_api/serviceaccount/#service-account-api)`,
+
+		CreateContext: stackServiceAccountTokenCreate,
+		ReadContext:   stackServiceAccountTokenRead,
+		DeleteContext: stackServiceAccountTokenDelete,
+
+		Schema: map[string]*schema.Schema{
+			"stack_slug": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"service_account_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"seconds_to_live": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				ForceNew: true,
+			},
+			"key": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+			"expiration": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"has_expired": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func stackServiceAccountTokenCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	c, cleanup, err := getClientForSATokenManagement(d, m)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	defer cleanup()
+
+	serviceAccountID, err := strconv.ParseInt(d.Get("service_account_id").(string), 10, 64)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	name := d.Get("name").(string)
+	ttl := d.Get("seconds_to_live").(int)
+
+	request := gapi.CreateServiceAccountTokenRequest{
+		Name:             name,
+		ServiceAccountID: serviceAccountID,
+		SecondsToLive:    int64(ttl),
+	}
+	response, err := c.CreateServiceAccountToken(request)
```

Comment:
```suggestion
	response, err := c.CreateStackServiceAccountToken(request)
```

this imply that we need the client to be changed as well? 🤔

Reply:
**Edit**: sorry, I think I completely misunderstood your point in my first response. If you're saying the same that Julien is saying [here](https://github.com/grafana/terraform-provider-grafana/pull/858#issuecomment-1482107129), then it totally makes sense and yes, we should do it :)